subscriptions.go

```diff
@@ -10,14 +10,27 @@ import (
 	"github.com/pkg/errors"
 )
 
+// AuthorizationAttribute represents a record for SubscriptionAuthorization which is used by the Nakadi high level API.
+type AuthorizationAttribute struct {
+	DataType string `json:"data_type"`
+	Value    string `json:"value"`
+}
+
+// SubscriptionAuthorization represents a subscription auth as used by the Nakadi high level API.
+type SubscriptionAuthorization struct {
+	Admins  []AuthorizationAttribute `json:"admins"`
+	Readers []AuthorizationAttribute `json:"readers"`
+}
+
 // Subscription represents a subscription as used by the Nakadi high level API.
 type Subscription struct {
-	ID                string    `json:"id,omitempty"`
-	OwningApplication string    `json:"owning_application"`
-	EventTypes        []string  `json:"event_types"`
-	ConsumerGroup     string    `json:"consumer_group,omitempty"`
-	ReadFrom          string    `json:"read_from,omitempty"`
-	CreatedAt         time.Time `json:"created_at,omitempty"`
+	ID                string                    `json:"id,omitempty"`
+	OwningApplication string                    `json:"owning_application"`
+	EventTypes        []string                  `json:"event_types"`
+	ConsumerGroup     string                    `json:"consumer_group,omitempty"`
+	ReadFrom          string                    `json:"read_from,omitempty"`
+	CreatedAt         time.Time                 `json:"created_at,omitempty"`
+	Authorization     SubscriptionAuthorization `json:"authorization"`
```

Comment:
According to the documentation the[ subscription authorization](https://nakadi.io/manual.html#definition_Subscription) is optional and can be missing. Therefore I suggest to change `Authorization` to a pointer `*SubscriptionAuthorization` and to change the json tag to json:"authorization,omitempty"`

Reply:
@stoewer 
There are 2 approaches to get subscription id:
1. Hardcode it
2. Use method for subscription creation with the same parameters (event types, owning app, consumer group) to get the same subscription id.
For the latter case subscription authorization obviously should be optional, as services usually do not know anything about latest authorization fixes, the only thing they care about is to get subscription id.